mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go

```diff
@@ -426,22 +429,26 @@ func cloudBackupScheduleCreateOrUpdate(ctx context.Context, conn *matlas.Client,
 		policiesItem = append(policiesItem, policyItem)
 	}
 	if v, ok := d.GetOk("policy_item_weekly"); ok {
-		item := v.([]interface{})
-		itemObj := item[0].(map[string]interface{})
-		policyItem.FrequencyType = snapshotScheduleWeekly
-		policyItem.RetentionUnit = itemObj["retention_unit"].(string)
-		policyItem.FrequencyInterval = itemObj["frequency_interval"].(int)
-		policyItem.RetentionValue = itemObj["retention_value"].(int)
-		policiesItem = append(policiesItem, policyItem)
+		items := v.([]interface{})
+		for _, s := range items {
+			itemObj := s.(map[string]interface{})
+			policyItem.FrequencyType = snapshotScheduleWeekly
+			policyItem.RetentionUnit = itemObj["retention_unit"].(string)
+			policyItem.FrequencyInterval = itemObj["frequency_interval"].(int)
+			policyItem.RetentionValue = itemObj["retention_value"].(int)
+			policiesItem = append(policiesItem, policyItem)
```

Comment:
[nit/not blocking] This logic is repeated in a few places - it would make sense to make a method that returns the modified item, so you can reuse and change the code in a single place.